src/compiler/diagnosticMessages.json

```diff
@@ -939,6 +939,14 @@
         "category": "Error",
         "code": 1335
     },
+    "An index signature parameter type cannot be a type alias. Use '[index: {0}]' instead.": {
+        "category": "Error",
+        "code": 1336
+    },
+    "An index signature parameter type cannot be a union type. Use '[K in {0}]' instead.": {

```

Comment:
Nit: this won't work if there are any other members in the object type. Maybe we should just write "Consider using a mapped object type instead.

Reply:
This is supposed to be thrown only if all members are string literals. How about changing the wording to `A union of string literals cannot be used as an index signature parameter type directly. Use '[K in {0}]' instead.`?